### ISSUE TITLE
Can I add batch norm to alexnet and load pre-trained weights?

### ISSUE THREAD

**Issue by wangg12** [kratzert/finetune_alexnet_with_tensorflow#1]

**Comment by kratzert** [comment]
You can and I do exactly the same for the VGG model in [this](https://github.com/kratzert/EGU2017_public/blob/master/examples/model.py) implementation. The only thing you have to take care of is, that you use BatchNorm only from before the first layer you start training from scratch. You can't use BatchNorm in between layers you want to initialize with pretrained weights, since then you change the activations and the pretrained weights don't fit to the actual activations anymore.

But for example if you train FC7 and FC8 from scratch then you can add BatchNorm after FC6 and then of course after FC7 as well.

I actually wanted to write and blog post about this possibility as well, since it helps to increase the accuracy of your finetuned network quite a bit. Might do this soon.

**Comment by wangg12** [comment]
@kratzert Thanks.
